src/cpp/src/continuous_batching_impl.cpp

```diff
@@ -168,9 +167,8 @@ void ContinuousBatchingPipeline::ContinuousBatchingImpl::step() {
             }
         }
         _free_non_running_requests();
-        return;
     }
-
+    m_pipeline_metrics.total_num_scheduled_tokens = scheduler_output.m_total_num_scheduled_tokens;
```

Comment:
can you move to a place where other `m_pipeline_metrics` values are filled?

Reply:
done